In [28]:
import pandas as pd


In [17]:
data=pd.read_excel('/content/iris (1).xls')
data

SL   SW   PL   PW  Classification
0    5.1  3.5  1.4  0.2     Iris-setosa
1    4.9  3.0  1.4  0.2     Iris-setosa
2    4.7  3.2  1.3  0.2     Iris-setosa
3    4.6  3.1  1.5  0.2     Iris-setosa
4    5.0  3.6  1.4  0.2     Iris-setosa
..   ...  ...  ...  ...             ...
145  6.7  3.0  5.2  2.3  Iris-virginica
146  6.3  2.5  5.0  1.9  Iris-virginica
147  6.5  3.0  5.2  2.0  Iris-virginica
148  6.2  3.4  5.4  2.3  Iris-virginica
149  5.9  3.0  5.1  1.8  Iris-virginica

[150 rows x 5 columns]

In [18]:
#calculating mean for each column

data['SL'].mean()

5.843333333333334

In [20]:
data['SW'].mean()

3.0540000000000003

In [21]:
data['PL'].mean()

3.758666666666666

In [22]:
data['PW'].mean()

1.1986666666666668

In [23]:
# Checking null values in the dataset

data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   SL              150 non-null    float64
 1   SW              150 non-null    float64
 2   PL              150 non-null    float64
 3   PW              150 non-null    float64
 4   Classification  150 non-null    object 
dtypes: float64(4), object(1)
memory usage: 6.0+ KB


In [27]:
data.isna().sum()

SL                0
SW                0
PL                0
PW                0
Classification    0
dtype: int64

# Insights from the data

1. From the average value of Sepal length, Sepal width, Petal length and Petal width, Sepal length is greater than others
2. In this 150 database,
Iris-setosa contains  50 samples,
Iris-versicolor contains 50 samples and
Iris-virginica contains 50 samples
3.  Sepal length- Min: 4.3,  Max: 7.9,  Range= 3.6;
    Sepal width-  Min: 2.0,  Max: 4.4,  Range= 2.4;
    Petal length- Min: 1.0,  Max: 6.9,  Range= 5.9;
    Petal width-  Min: 0.1,  Max: 2.5,  Range= 2.4